diff --git a/src/core/mount.c b/src/core/mount.c
index de1075d..877cd8b 100644
--- a/src/core/mount.c
+++ b/src/core/mount.c
@@ -1450,7 +1450,7 @@ static int mount_setup_unit(
 
         if (set_flags) {
                 MOUNT(u)->is_mounted = true;
-                MOUNT(u)->just_mounted = !MOUNT(u)->from_proc_self_mountinfo;
+                MOUNT(u)->just_mounted = !MOUNT(u)->from_proc_self_mountinfo || MOUNT(u)->just_mounted;
                 MOUNT(u)->just_changed = changed;
         }
 
